metron-platform/metron-writer/src/main/java/org/apache/metron/writer/bolt/BulkMessageWriterBolt.java

```diff
@@ -92,24 +177,58 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
       configurationTransformation = x -> x;
     }
     try {
-      bulkMessageWriter.init(stormConf,
-                             context,
-                             configurationTransformation.apply(new IndexingWriterConfiguration(bulkMessageWriter.getName(),
-                             getConfigurations()))
-                            );
+      WriterConfiguration writerconf = configurationTransformation.apply(
+              new IndexingWriterConfiguration(bulkMessageWriter.getName(), getConfigurations()));
+      if (defaultBatchTimeout == 0) {
+        //This means getComponentConfiguration was never called to initialize defaultBatchTimeout,
+        //probably because we are in a unit test scenario.  So calculate it here.
+        BatchTimeoutHelper timeoutHelper = new BatchTimeoutHelper(writerconf::getAllConfiguredTimeouts, batchTimeoutDivisor);
+        defaultBatchTimeout = timeoutHelper.getDefaultBatchTimeout();
+      }
+      writerComponent.setDefaultBatchTimeout(defaultBatchTimeout);
+      bulkMessageWriter.init(stormConf, context, writerconf);
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Used only for unit testing.
+   */
```

Comment:
Maybe we can name this testPrepare? So it is explicit to caller, who may not 'peek' inside the source